net/discovery.go

```diff
@@ -27,50 +27,12 @@ type discovery struct {
 	ctx         context.Context
 	dht         *dual.DHT
 	h           libp2phost.Host
-	rd          *libp2pdiscovery.RoutingDiscovery
-	provides    []types.ProvidesCoin
-	advertiseCh chan struct{}
+	rd          *libp2prouting.RoutingDiscovery
+	provides    []types.ProvidesCoin // set to a single item slice of XMR when we make an offer
+	advertiseCh chan struct{}        // signals to advertise now that an XMR offer was made
 	offerAPI    Handler
 }
 
-func newDiscovery(
-	ctx context.Context,
-	h libp2phost.Host,
-	bnsFunc func() []peer.AddrInfo,
-) (*discovery, error) {
-	dhtOpts := []dual.Option{
-		dual.DHTOption(kaddht.BootstrapPeersFunc(bnsFunc)),
-		dual.DHTOption(kaddht.Mode(kaddht.ModeAutoServer)),
-	}
-
-	// There is libp2p bug when calling `dual.New` with a cancelled context creating a panic,
-	// so we added the extra guard below:
-	// Panic:  https://github.com/jbenet/goprocess/blob/v0.1.4/impl-mutex.go#L99
-	// Caller: https://github.com/libp2p/go-libp2p-kad-dht/blob/v0.17.0/dht.go#L222
-	//
-	select {
-	case <-ctx.Done():
-		return nil, ctx.Err()
-	default:
-		// not cancelled, continue on
-	}
-
-	dht, err := dual.New(ctx, h, dhtOpts...)
-	if err != nil {
-		return nil, err
-	}
```

Comment:
The code in this function was moved to host.go, as the `dht` created above is needed to initialize libp2p's routedhost.